<a href="https://www.kaggle.com/code/eugenetanake/data-cleaning-flights-pt2-understand-status?scriptVersionId=110937851" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

In [1]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

/kaggle/input/data-cleaning-flights-pt1-initial-clean/__results__.html
/kaggle/input/data-cleaning-flights-pt1-initial-clean/__resultx__.html
/kaggle/input/data-cleaning-flights-pt1-initial-clean/__notebook__.ipynb
/kaggle/input/data-cleaning-flights-pt1-initial-clean/__output__.json
/kaggle/input/data-cleaning-flights-pt1-initial-clean/flights_cleaned.csv
/kaggle/input/data-cleaning-flights-pt1-initial-clean/custom.css


In [2]:
df = pd.read_csv('/kaggle/input/data-cleaning-flights-pt1-initial-clean/flights_cleaned.csv')

In [3]:
df.columns

Index(['number', 'status', 'codeshareStatus', 'departure.scheduledTimeLocal',
       'departure.actualTimeLocal', 'departure.terminal',
       'departure.checkInDesk', 'departure.gate', 'arrival.airport.icao',
       'arrival.airport.iata', 'arrival.airport.name', 'airline.name'],
      dtype='object')

In [4]:
df.isnull().sum()

number                            0
status                            0
codeshareStatus                   0
departure.scheduledTimeLocal      0
departure.actualTimeLocal       353
departure.terminal               21
departure.checkInDesk           668
departure.gate                  766
arrival.airport.icao            200
arrival.airport.iata            200
arrival.airport.name              0
airline.name                      0
dtype: int64

In [5]:
df.head()

number    status codeshareStatus departure.scheduledTimeLocal  \
0  A3 1207  Departed    IsCodeshared       2022-06-06 00:30+08:00   
1  AC 6307  Departed    IsCodeshared       2022-06-06 00:10+08:00   
2   CA 642  Departed    IsCodeshared       2022-06-06 00:20+08:00   
3  CAO 642  Departed      IsOperator       2022-06-06 00:20+08:00   
4  LH 9754  Departed    IsCodeshared       2022-06-06 00:45+08:00   

  departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
0    2022-06-06 00:30+08:00                 3.0                    3.0   
1    2022-06-06 00:10+08:00                 3.0                    3.0   
2    2022-06-06 00:20+08:00                 NaN                    NaN   
3    2022-06-06 00:20+08:00                 NaN                    NaN   
4    2022-06-06 00:45+08:00                 3.0                    3.0   

  departure.gate arrival.airport.icao arrival.airport.iata  \
0            A17                 EDDM                  MUC   
1             B9                 RKSI                  ICN   
2            NaN                 ZSPD                  PVG   
3            NaN                 ZSPD                  PVG   
4             A4                 YSSY                  SYD   

  arrival.airport.name   airline.name  
0               Munich         Aegean  
1                Seoul     Air Canada  
2             Shanghai  Air China LTD  
3             Shanghai            CAO  
4          Sydney City      Lufthansa

The two columns which requires understanding the most, would be the "status" and the "IsCodeshareStatus" column.

In [6]:
df['status'].value_counts()

Departed             5461
Expected              782
Boarding               47
CanceledUncertain      29
Canceled               25
GateClosed             17
Delayed                11
Unknown                 9
Name: status, dtype: int64

In [7]:
df['codeshareStatus'].value_counts()

IsCodeshared    4086
IsOperator      2278
Unknown           17
Name: codeshareStatus, dtype: int64

In [8]:
df[df['status']=='CanceledUncertain']

number             status codeshareStatus departure.scheduledTimeLocal  \
139   FD 358  CanceledUncertain         Unknown       2022-06-06 10:15+08:00   
360   MH 608  CanceledUncertain      IsOperator       2022-06-06 18:10+08:00   
454   QZ 269  CanceledUncertain         Unknown       2022-06-06 21:55+08:00   
815   FD 358  CanceledUncertain         Unknown       2022-06-07 10:15+08:00   
1007  MH 632  CanceledUncertain         Unknown       2022-06-07 14:50+08:00   
1084  MH 616  CanceledUncertain      IsOperator       2022-06-07 19:15+08:00   
1208  QR 947  CanceledUncertain      IsOperator       2022-06-07 21:25+08:00   
1473  FD 358  CanceledUncertain         Unknown       2022-06-08 10:15+08:00   
1719  MH 608  CanceledUncertain      IsOperator       2022-06-08 18:10+08:00   
1746  MH 616  CanceledUncertain      IsOperator       2022-06-08 19:15+08:00   
1799  QZ 269  CanceledUncertain         Unknown       2022-06-08 21:55+08:00   
2164  FD 358  CanceledUncertain         Unknown       2022-06-09 10:15+08:00   
2189  VN 694  CanceledUncertain      IsOperator       2022-06-09 11:30+08:00   
2191  JT 163  CanceledUncertain      IsOperator       2022-06-09 12:00+08:00   
2400  MH 608  CanceledUncertain      IsOperator       2022-06-09 18:10+08:00   
2421  MH 616  CanceledUncertain      IsOperator       2022-06-09 19:15+08:00   
2485  QZ 269  CanceledUncertain         Unknown       2022-06-09 21:55+08:00   
2566  QR 947  CanceledUncertain      IsOperator       2022-06-09 21:25+08:00   
2831  FD 358  CanceledUncertain         Unknown       2022-06-10 10:15+08:00   
2896  CI 756  CanceledUncertain      IsOperator       2022-06-10 02:00+08:00   
3114  MH 616  CanceledUncertain      IsOperator       2022-06-10 19:15+08:00   
3177  QZ 269  CanceledUncertain         Unknown       2022-06-10 21:55+08:00   
3315  VZ 621  CanceledUncertain         Unknown       2022-06-10 14:10+08:00   
3432  TR 840  CanceledUncertain         Unknown       2022-06-11 01:25+08:00   
3556  FD 358  CanceledUncertain         Unknown       2022-06-11 10:15+08:00   
3795  MH 608  CanceledUncertain      IsOperator       2022-06-11 18:10+08:00   
3825  MH 616  CanceledUncertain      IsOperator       2022-06-11 19:15+08:00   
3892  QR 947  CanceledUncertain      IsOperator       2022-06-11 21:25+08:00   
3893  QZ 269  CanceledUncertain         Unknown       2022-06-11 21:55+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
139                        NaN                 1.0                    NaN   
360                        NaN                 1.0                    NaN   
454                        NaN                 1.0                    NaN   
815                        NaN                 1.0                    NaN   
1007                       NaN                 1.0                    NaN   
1084                       NaN                 1.0                    NaN   
1208                       NaN                 1.0                    NaN   
1473                       NaN                 1.0                    NaN   
1719                       NaN                 1.0                    NaN   
1746                       NaN                 1.0                    NaN   
1799                       NaN                 1.0                    NaN   
2164                       NaN                 1.0                    NaN   
2189                       NaN                 3.0                    NaN   
2191    2022-06-09 12:00+08:00                 3.0                    1.0   
2400                       NaN                 1.0                    NaN   
2421                       NaN                 1.0                    NaN   
2485                       NaN                 1.0                    NaN   
2566                       NaN                 1.0                    NaN   
2831                       NaN                 1.0                    NaN   
2896                       NaN                 3.0               

From their website:
https://aerodatabox.com/adding-usa-live-flights-coverage-adding-mode-s-transponder-code-covid-19-changes-airport-daily-statistic-routes-update/

CanceledUncertain Flights, as stated by their website, are for flights that simply "disappear" off the lists and for other ambiguous reasons. There is still a chance that the flight has taken off. As seen on row 2191, there is a record of the flight taking off even with status "CanceledUncertain". I'm going to look deeper into each individual number.

In [9]:
df[df['status']=='CanceledUncertain']['number'].value_counts()

FD 358    6
QZ 269    5
MH 616    5
MH 608    4
QR 947    3
MH 632    1
VN 694    1
JT 163    1
CI 756    1
VZ 621    1
TR 840    1
Name: number, dtype: int64

In [10]:
df[df['number']=='FD 358']

number             status codeshareStatus departure.scheduledTimeLocal  \
139   FD 358  CanceledUncertain         Unknown       2022-06-06 10:15+08:00   
577   FD 358           Departed      IsOperator       2022-06-06 17:10+08:00   
815   FD 358  CanceledUncertain         Unknown       2022-06-07 10:15+08:00   
1247  FD 358           Departed      IsOperator       2022-06-07 17:10+08:00   
1473  FD 358  CanceledUncertain         Unknown       2022-06-08 10:15+08:00   
1931  FD 358           Departed      IsOperator       2022-06-08 17:10+08:00   
2164  FD 358  CanceledUncertain         Unknown       2022-06-09 10:15+08:00   
2611  FD 358           Departed      IsOperator       2022-06-09 17:10+08:00   
2831  FD 358  CanceledUncertain         Unknown       2022-06-10 10:15+08:00   
3313  FD 358           Departed      IsOperator       2022-06-10 17:10+08:00   
3556  FD 358  CanceledUncertain         Unknown       2022-06-11 10:15+08:00   
4008  FD 358           Departed      IsOperator       2022-06-11 17:10+08:00   
4722  FD 358           Departed      IsOperator       2022-06-12 17:10+08:00   
5454  FD 358           Expected      IsOperator       2022-06-13 17:10+08:00   
6300  FD 358           Departed      IsOperator       2022-06-13 17:10+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
139                        NaN                 1.0                    NaN   
577     2022-06-06 17:10+08:00                 1.0                    2.0   
815                        NaN                 1.0                    NaN   
1247    2022-06-07 17:10+08:00                 1.0                    2.0   
1473                       NaN                 1.0                    NaN   
1931    2022-06-08 17:10+08:00                 1.0                    2.0   
2164                       NaN                 1.0                    NaN   
2611    2022-06-09 17:10+08:00                 1.0                    2.0   
2831                       NaN                 1.0                    NaN   
3313    2022-06-10 17:10+08:00                 1.0                    2.0   
3556                       NaN                 1.0                    NaN   
4008    2022-06-11 17:10+08:00                 1.0                    2.0   
4722    2022-06-12 17:10+08:00                 1.0                    2.0   
5454    2022-06-13 17:10+08:00                 1.0                    NaN   
6300    2022-06-13 17:10+08:00                 1.0                    2.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
139             NaN                  NaN                  NaN   
577             C11                 VTBD                  DMK   
815             NaN                  NaN                  NaN   
1247            D38                 VTBD                  DMK   
1473            NaN                  NaN                  NaN   
1931            C17                 VTBD                  DMK   
2164            NaN                  NaN                  NaN   
2611            D36                 VTBD                  DMK   
2831            NaN                  NaN                  NaN   
3313            D36                 VTBD                  DMK   
3556            NaN                  NaN                  NaN   
4008            C17                 VTBD                  DMK   
4722            D30                 VTBD                  DMK   
5454            NaN                 VTBD                  DMK   
6300            D47                 VTBD                  DMK   

     arrival.airport.name  airline.name  
139               Bangkok  Thai AirAsia  
577               Bangkok  Thai AirAsia  
815               Bangkok  Thai AirAsia  
1247              Bangkok  Thai AirAsia  
1473              Bangkok  Thai AirAsia  
1931              Bangkok  Thai AirAsia  
2164              Bangkok  Thai AirAsia  
2611              Bangkok  Thai AirAsia  
2831              Bangkok  Thai AirAsia  
3313              Bangkok  Thai

In [11]:
list(df[df['status']=='CanceledUncertain']['number'].value_counts().index)

['FD 358',
 'QZ 269',
 'MH 616',
 'MH 608',
 'QR 947',
 'MH 632',
 'VN 694',
 'JT 163',
 'CI 756',
 'VZ 621',
 'TR 840']

In [12]:
for i in list(df[df['status']=='CanceledUncertain']['number'].value_counts().index):
    #display(df[df['number']==i].groupby('departure.scheduledTimeLocal').apply(display))
    display(df[df['number']==i])

number             status codeshareStatus departure.scheduledTimeLocal  \
139   FD 358  CanceledUncertain         Unknown       2022-06-06 10:15+08:00   
577   FD 358           Departed      IsOperator       2022-06-06 17:10+08:00   
815   FD 358  CanceledUncertain         Unknown       2022-06-07 10:15+08:00   
1247  FD 358           Departed      IsOperator       2022-06-07 17:10+08:00   
1473  FD 358  CanceledUncertain         Unknown       2022-06-08 10:15+08:00   
1931  FD 358           Departed      IsOperator       2022-06-08 17:10+08:00   
2164  FD 358  CanceledUncertain         Unknown       2022-06-09 10:15+08:00   
2611  FD 358           Departed      IsOperator       2022-06-09 17:10+08:00   
2831  FD 358  CanceledUncertain         Unknown       2022-06-10 10:15+08:00   
3313  FD 358           Departed      IsOperator       2022-06-10 17:10+08:00   
3556  FD 358  CanceledUncertain         Unknown       2022-06-11 10:15+08:00   
4008  FD 358           Departed      IsOperator       2022-06-11 17:10+08:00   
4722  FD 358           Departed      IsOperator       2022-06-12 17:10+08:00   
5454  FD 358           Expected      IsOperator       2022-06-13 17:10+08:00   
6300  FD 358           Departed      IsOperator       2022-06-13 17:10+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
139                        NaN                 1.0                    NaN   
577     2022-06-06 17:10+08:00                 1.0                    2.0   
815                        NaN                 1.0                    NaN   
1247    2022-06-07 17:10+08:00                 1.0                    2.0   
1473                       NaN                 1.0                    NaN   
1931    2022-06-08 17:10+08:00                 1.0                    2.0   
2164                       NaN                 1.0                    NaN   
2611    2022-06-09 17:10+08:00                 1.0                    2.0   
2831                       NaN                 1.0                    NaN   
3313    2022-06-10 17:10+08:00                 1.0                    2.0   
3556                       NaN                 1.0                    NaN   
4008    2022-06-11 17:10+08:00                 1.0                    2.0   
4722    2022-06-12 17:10+08:00                 1.0                    2.0   
5454    2022-06-13 17:10+08:00                 1.0                    NaN   
6300    2022-06-13 17:10+08:00                 1.0                    2.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
139             NaN                  NaN                  NaN   
577             C11                 VTBD                  DMK   
815             NaN                  NaN                  NaN   
1247            D38                 VTBD                  DMK   
1473            NaN                  NaN                  NaN   
1931            C17                 VTBD                  DMK   
2164            NaN                  NaN                  NaN   
2611            D36                 VTBD                  DMK   
2831            NaN                  NaN                  NaN   
3313            D36                 VTBD                  DMK   
3556            NaN                  NaN                  NaN   
4008            C17                 VTBD                  DMK   
4722            D30                 VTBD                  DMK   
5454            NaN                 VTBD                  DMK   
6300            D47                 VTBD                  DMK   

     arrival.airport.name  airline.name  
139               Bangkok  Thai AirAsia  
577               Bangkok  Thai AirAsia  
815               Bangkok  Thai AirAsia  
1247              Bangkok  Thai AirAsia  
1473              Bangkok  Thai AirAsia  
1931              Bangkok  Thai AirAsia  
2164              Bangkok  Thai AirAsia  
2611              Bangkok  Thai AirAsia  
2831              Bangkok  Thai AirAsia  
3313              Bangkok  Thai

number             status codeshareStatus departure.scheduledTimeLocal  \
454   QZ 269  CanceledUncertain         Unknown       2022-06-06 21:55+08:00   
1129  QZ 269           Departed      IsOperator       2022-06-07 21:55+08:00   
1799  QZ 269  CanceledUncertain         Unknown       2022-06-08 21:55+08:00   
2485  QZ 269  CanceledUncertain         Unknown       2022-06-09 21:55+08:00   
3177  QZ 269  CanceledUncertain         Unknown       2022-06-10 21:55+08:00   
3893  QZ 269  CanceledUncertain         Unknown       2022-06-11 21:55+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
454                        NaN                 1.0                    NaN   
1129    2022-06-07 21:55+08:00                 1.0                    NaN   
1799                       NaN                 1.0                    NaN   
2485                       NaN                 1.0                    NaN   
3177                       NaN                 1.0                    NaN   
3893                       NaN                 1.0                    NaN   

     departure.gate arrival.airport.icao arrival.airport.iata  \
454             NaN                  NaN                  NaN   
1129            NaN                  NaN                  NaN   
1799            NaN                  NaN                  NaN   
2485            NaN                  NaN                  NaN   
3177            NaN                  NaN                  NaN   
3893            NaN                  NaN                  NaN   

     arrival.airport.name       airline.name  
454               Jakarta  Indonesia AirAsia  
1129              Jakarta  Indonesia AirAsia  
1799              Jakarta  Indonesia AirAsia  
2485              Jakarta  Indonesia AirAsia  
3177              Jakarta  Indonesia AirAsia  
3893              Jakarta  Indonesia AirAsia

number             status codeshareStatus departure.scheduledTimeLocal  \
1084  MH 616  CanceledUncertain      IsOperator       2022-06-07 19:15+08:00   
1746  MH 616  CanceledUncertain      IsOperator       2022-06-08 19:15+08:00   
2421  MH 616  CanceledUncertain      IsOperator       2022-06-09 19:15+08:00   
3114  MH 616  CanceledUncertain      IsOperator       2022-06-10 19:15+08:00   
3825  MH 616  CanceledUncertain      IsOperator       2022-06-11 19:15+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
1084                       NaN                 1.0                    NaN   
1746                       NaN                 1.0                    NaN   
2421                       NaN                 1.0                    NaN   
3114                       NaN                 1.0                    NaN   
3825                       NaN                 1.0                    NaN   

     departure.gate arrival.airport.icao arrival.airport.iata  \
1084            NaN                  NaN                  NaN   
1746            NaN                  NaN                  NaN   
2421            NaN                  NaN                  NaN   
3114            NaN                  NaN                  NaN   
3825            NaN                  NaN                  NaN   

     arrival.airport.name airline.name  
1084         Kuala Lumpur     Malaysia  
1746         Kuala Lumpur     Malaysia  
2421         Kuala Lumpur     Malaysia  
3114         Kuala Lumpur     Malaysia  
3825         Kuala Lumpur     Malaysia

number             status codeshareStatus departure.scheduledTimeLocal  \
360   MH 608  CanceledUncertain      IsOperator       2022-06-06 18:10+08:00   
1061  MH 608           Canceled      IsOperator       2022-06-07 18:10+08:00   
1719  MH 608  CanceledUncertain      IsOperator       2022-06-08 18:10+08:00   
2400  MH 608  CanceledUncertain      IsOperator       2022-06-09 18:10+08:00   
3076  MH 608           Departed      IsOperator       2022-06-10 18:10+08:00   
3795  MH 608  CanceledUncertain      IsOperator       2022-06-11 18:10+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
360                        NaN                 1.0                    NaN   
1061    2022-06-07 18:10+08:00                 1.0                   12.0   
1719                       NaN                 1.0                    NaN   
2400                       NaN                 1.0                    NaN   
3076                       NaN                 1.0                    NaN   
3795                       NaN                 1.0                    NaN   

     departure.gate arrival.airport.icao arrival.airport.iata  \
360             NaN                  NaN                  NaN   
1061            C16                 WMKK                  KUL   
1719            NaN                  NaN                  NaN   
2400            NaN                  NaN                  NaN   
3076            NaN                  NaN                  NaN   
3795            NaN                  NaN                  NaN   

     arrival.airport.name airline.name  
360          Kuala Lumpur     Malaysia  
1061         Kuala Lumpur     Malaysia  
1719         Kuala Lumpur     Malaysia  
2400         Kuala Lumpur     Malaysia  
3076         Kuala Lumpur     Malaysia  
3795         Kuala Lumpur     Malaysia

number             status codeshareStatus departure.scheduledTimeLocal  \
821   QR 947           Departed      IsOperator       2022-06-07 10:35+08:00   
1208  QR 947  CanceledUncertain      IsOperator       2022-06-07 21:25+08:00   
2169  QR 947           Departed      IsOperator       2022-06-09 10:35+08:00   
2566  QR 947  CanceledUncertain      IsOperator       2022-06-09 21:25+08:00   
3616  QR 947           Departed      IsOperator       2022-06-11 10:35+08:00   
3892  QR 947  CanceledUncertain      IsOperator       2022-06-11 21:25+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
821     2022-06-07 10:35+08:00                 1.0                    3.0   
1208                       NaN                 1.0                    NaN   
2169    2022-06-09 10:35+08:00                 1.0                    1.0   
2566                       NaN                 1.0                    NaN   
3616    2022-06-11 10:35+08:00                 1.0                    1.0   
3892                       NaN                 1.0                    NaN   

     departure.gate arrival.airport.icao arrival.airport.iata  \
821             D34                 OTHH                  DOH   
1208            NaN                 OTHH                  DOH   
2169            C22                 OTHH                  DOH   
2566            NaN                 OTHH                  DOH   
3616            D32                 OTHH                  DOH   
3892            NaN                  NaN                  NaN   

     arrival.airport.name   airline.name  
821                  Doha  Qatar Airways  
1208                 Doha  Qatar Airways  
2169                 Doha  Qatar Airways  
2566                 Doha  Qatar Airways  
3616                 Doha  Qatar Airways  
3892                 Doha  Qatar Airways

number             status codeshareStatus departure.scheduledTimeLocal  \
1007  MH 632  CanceledUncertain         Unknown       2022-06-07 14:50+08:00   
2668  MH 632           Departed      IsOperator       2022-06-09 14:50+08:00   
4063  MH 632           Departed      IsOperator       2022-06-11 14:50+08:00   
4786  MH 632           Departed      IsOperator       2022-06-12 14:50+08:00   
6011  MH 632           Canceled      IsOperator       2022-06-13 14:50+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
1007                       NaN                 1.0                    NaN   
2668    2022-06-09 14:50+08:00                 1.0                   12.0   
4063    2022-06-11 14:50+08:00                 1.0                   12.0   
4786    2022-06-12 14:50+08:00                 1.0                   12.0   
6011    2022-06-13 14:50+08:00                 1.0                    NaN   

     departure.gate arrival.airport.icao arrival.airport.iata  \
1007            NaN                  NaN                  NaN   
2668            D36                 WMKP                  PEN   
4063             C1                 WMKP                  PEN   
4786            D32                 WMKP                  PEN   
6011            NaN                 WMKP                  PEN   

     arrival.airport.name airline.name  
1007               Penang     Malaysia  
2668               Penang     Malaysia  
4063               Penang     Malaysia  
4786               Penang     Malaysia  
6011               Penang     Malaysia

number             status codeshareStatus departure.scheduledTimeLocal  \
845   VN 694           Canceled      IsOperator       2022-06-07 11:30+08:00   
2189  VN 694  CanceledUncertain      IsOperator       2022-06-09 11:30+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
845     2022-06-07 11:30+08:00                 3.0                    9.0   
2189                       NaN                 3.0                    NaN   

     departure.gate arrival.airport.icao arrival.airport.iata  \
845              B6                 VVPQ                  PQC   
2189            NaN                 VVPQ                  PQC   

     arrival.airport.name airline.name  
845            Duong Dong      Vietnam  
2189           Duong Dong      Vietnam

number             status codeshareStatus departure.scheduledTimeLocal  \
157   JT 163           Departed      IsOperator       2022-06-06 12:00+08:00   
847   JT 163           Canceled      IsOperator       2022-06-07 12:00+08:00   
1499  JT 163           Canceled      IsOperator       2022-06-08 12:00+08:00   
2191  JT 163  CanceledUncertain      IsOperator       2022-06-09 12:00+08:00   
2861  JT 163           Canceled      IsOperator       2022-06-10 12:00+08:00   
3582  JT 163           Canceled      IsOperator       2022-06-11 12:00+08:00   
4277  JT 163           Canceled      IsOperator       2022-06-12 12:00+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
157     2022-06-06 12:00+08:00                 3.0                    1.0   
847     2022-06-07 12:00+08:00                 3.0                    1.0   
1499    2022-06-08 12:00+08:00                 3.0                    1.0   
2191    2022-06-09 12:00+08:00                 3.0                    1.0   
2861    2022-06-10 12:00+08:00                 3.0                    1.0   
3582    2022-06-11 12:00+08:00                 3.0                    1.0   
4277    2022-06-12 12:00+08:00                 3.0                    1.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
157              B2                 WIII                  CGK   
847             A17                 WIII                  CGK   
1499            A17                 WIII                  CGK   
2191            A19                 WIII                  CGK   
2861            A21                 WIII                  CGK   
3582            NaN                 WIII                  CGK   
4277            A11                 WIII                  CGK   

     arrival.airport.name airline.name  
157               Jakarta     Lion Air  
847               Jakarta     Lion Air  
1499              Jakarta     Lion Air  
2191              Jakarta     Lion Air  
2861              Jakarta     Lion Air  
3582              Jakarta     Lion Air  
4277              Jakarta     Lion Air

number             status codeshareStatus departure.scheduledTimeLocal  \
190   CI 756            Unknown      IsOperator       2022-06-06 02:00+08:00   
879   CI 756            Unknown      IsOperator       2022-06-07 02:00+08:00   
1532  CI 756            Unknown      IsOperator       2022-06-08 02:00+08:00   
2223  CI 756            Unknown      IsOperator       2022-06-09 02:00+08:00   
2896  CI 756  CanceledUncertain      IsOperator       2022-06-10 02:00+08:00   
3618  CI 756            Unknown      IsOperator       2022-06-11 02:00+08:00   
4310  CI 756           Departed      IsOperator       2022-06-12 02:00+08:00   
5022  CI 756            Unknown      IsOperator       2022-06-13 02:00+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
190                        NaN                 3.0                    NaN   
879                        NaN                 3.0                    NaN   
1532                       NaN                 3.0                    NaN   
2223                       NaN                 3.0                    NaN   
2896                       NaN                 3.0                    NaN   
3618                       NaN                 3.0                    NaN   
4310                       NaN                 3.0                    NaN   
5022                       NaN                 3.0                    NaN   

     departure.gate arrival.airport.icao arrival.airport.iata  \
190             NaN                 RCTP                  TPE   
879             NaN                 RCTP                  TPE   
1532            NaN                 RCTP                  TPE   
2223            NaN                 RCTP                  TPE   
2896            NaN                 RCTP                  TPE   
3618            NaN                 RCTP                  TPE   
4310            NaN                 RCTP                  TPE   
5022            NaN                 RCTP                  TPE   

     arrival.airport.name airline.name  
190                Taipei        China  
879                Taipei        China  
1532               Taipei        China  
2223               Taipei        China  
2896               Taipei        China  
3618               Taipei        China  
4310               Taipei        China  
5022               Taipei        China

number             status codeshareStatus departure.scheduledTimeLocal  \
1251  VZ 621           Departed      IsOperator       2022-06-07 14:10+08:00   
2612  VZ 621           Departed      IsOperator       2022-06-09 14:10+08:00   
3315  VZ 621  CanceledUncertain         Unknown       2022-06-10 14:10+08:00   
4014  VZ 621           Departed      IsOperator       2022-06-11 14:10+08:00   
4726  VZ 621           Departed      IsOperator       2022-06-12 14:10+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
1251    2022-06-07 14:10+08:00                 3.0                    1.0   
2612    2022-06-09 14:10+08:00                 3.0                    1.0   
3315                       NaN                 3.0                    NaN   
4014    2022-06-11 14:10+08:00                 3.0                    1.0   
4726    2022-06-12 14:10+08:00                 3.0                    1.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
1251            A16                 VTBS                  BKK   
2612             B9                 VTBS                  BKK   
3315            NaN                 VTBS                  BKK   
4014            B10                 VTBS                  BKK   
4726            A21                 VTBS                  BKK   

     arrival.airport.name      airline.name  
1251              Bangkok  Thai Vietjet Air  
2612              Bangkok  Thai Vietjet Air  
3315              Bangkok  Thai Vietjet Air  
4014              Bangkok  Thai Vietjet Air  
4726              Bangkok  Thai Vietjet Air

number             status codeshareStatus departure.scheduledTimeLocal  \
886   TR 840            Unknown         Unknown       2022-06-07 01:25+08:00   
2231  TR 840           Departed         Unknown       2022-06-09 01:25+08:00   
3432  TR 840  CanceledUncertain         Unknown       2022-06-11 01:25+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
886                        NaN                 1.0                    NaN   
2231                       NaN                 1.0                    NaN   
3432                       NaN                 1.0                    NaN   

     departure.gate arrival.airport.icao arrival.airport.iata  \
886             NaN                 RKSI                  ICN   
2231            NaN                 RKSI                  ICN   
3432            NaN                  NaN                  NaN   

     arrival.airport.name airline.name  
886                 Seoul        Scoot  
2231                Seoul        Scoot  
3432             Seoul-si        Scoot

My conclusion for now is that these canceled uncertain flights are not duplicates and may provide more insight afterwards and would be kept as it is for now.

In [13]:
for i in list(df[df['status']=='Unknown']['number'].value_counts().index):
    display(df[df['number']==i])

number             status codeshareStatus departure.scheduledTimeLocal  \
190   CI 756            Unknown      IsOperator       2022-06-06 02:00+08:00   
879   CI 756            Unknown      IsOperator       2022-06-07 02:00+08:00   
1532  CI 756            Unknown      IsOperator       2022-06-08 02:00+08:00   
2223  CI 756            Unknown      IsOperator       2022-06-09 02:00+08:00   
2896  CI 756  CanceledUncertain      IsOperator       2022-06-10 02:00+08:00   
3618  CI 756            Unknown      IsOperator       2022-06-11 02:00+08:00   
4310  CI 756           Departed      IsOperator       2022-06-12 02:00+08:00   
5022  CI 756            Unknown      IsOperator       2022-06-13 02:00+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
190                        NaN                 3.0                    NaN   
879                        NaN                 3.0                    NaN   
1532                       NaN                 3.0                    NaN   
2223                       NaN                 3.0                    NaN   
2896                       NaN                 3.0                    NaN   
3618                       NaN                 3.0                    NaN   
4310                       NaN                 3.0                    NaN   
5022                       NaN                 3.0                    NaN   

     departure.gate arrival.airport.icao arrival.airport.iata  \
190             NaN                 RCTP                  TPE   
879             NaN                 RCTP                  TPE   
1532            NaN                 RCTP                  TPE   
2223            NaN                 RCTP                  TPE   
2896            NaN                 RCTP                  TPE   
3618            NaN                 RCTP                  TPE   
4310            NaN                 RCTP                  TPE   
5022            NaN                 RCTP                  TPE   

     arrival.airport.name airline.name  
190                Taipei        China  
879                Taipei        China  
1532               Taipei        China  
2223               Taipei        China  
2896               Taipei        China  
3618               Taipei        China  
4310               Taipei        China  
5022               Taipei        China

number             status codeshareStatus departure.scheduledTimeLocal  \
886   TR 840            Unknown         Unknown       2022-06-07 01:25+08:00   
2231  TR 840           Departed         Unknown       2022-06-09 01:25+08:00   
3432  TR 840  CanceledUncertain         Unknown       2022-06-11 01:25+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
886                        NaN                 1.0                    NaN   
2231                       NaN                 1.0                    NaN   
3432                       NaN                 1.0                    NaN   

     departure.gate arrival.airport.icao arrival.airport.iata  \
886             NaN                 RKSI                  ICN   
2231            NaN                 RKSI                  ICN   
3432            NaN                  NaN                  NaN   

     arrival.airport.name airline.name  
886                 Seoul        Scoot  
2231                Seoul        Scoot  
3432             Seoul-si        Scoot

number   status codeshareStatus departure.scheduledTimeLocal  \
955  KE 363  Unknown      IsOperator       2022-06-07 05:30+08:00   

    departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
955                       NaN                 NaN                    NaN   

    departure.gate arrival.airport.icao arrival.airport.iata  \
955            NaN                 WMKP                  PEN   

    arrival.airport.name airline.name  
955               Penang   Korean Air

number   status codeshareStatus departure.scheduledTimeLocal  \
2284  KE 8383  Unknown         Unknown       2022-06-09 05:40+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
2284                       NaN                 NaN                    NaN   

     departure.gate arrival.airport.icao arrival.airport.iata  \
2284            NaN                 WMKK                  KUL   

     arrival.airport.name airline.name  
2284         Kuala Lumpur   Korean Air

In [14]:
for i in list(df[df['status']=='Canceled']['number'].value_counts().index):
    display(df[df['number']==i])

number             status codeshareStatus departure.scheduledTimeLocal  \
157   JT 163           Departed      IsOperator       2022-06-06 12:00+08:00   
847   JT 163           Canceled      IsOperator       2022-06-07 12:00+08:00   
1499  JT 163           Canceled      IsOperator       2022-06-08 12:00+08:00   
2191  JT 163  CanceledUncertain      IsOperator       2022-06-09 12:00+08:00   
2861  JT 163           Canceled      IsOperator       2022-06-10 12:00+08:00   
3582  JT 163           Canceled      IsOperator       2022-06-11 12:00+08:00   
4277  JT 163           Canceled      IsOperator       2022-06-12 12:00+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
157     2022-06-06 12:00+08:00                 3.0                    1.0   
847     2022-06-07 12:00+08:00                 3.0                    1.0   
1499    2022-06-08 12:00+08:00                 3.0                    1.0   
2191    2022-06-09 12:00+08:00                 3.0                    1.0   
2861    2022-06-10 12:00+08:00                 3.0                    1.0   
3582    2022-06-11 12:00+08:00                 3.0                    1.0   
4277    2022-06-12 12:00+08:00                 3.0                    1.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
157              B2                 WIII                  CGK   
847             A17                 WIII                  CGK   
1499            A17                 WIII                  CGK   
2191            A19                 WIII                  CGK   
2861            A21                 WIII                  CGK   
3582            NaN                 WIII                  CGK   
4277            A11                 WIII                  CGK   

     arrival.airport.name airline.name  
157               Jakarta     Lion Air  
847               Jakarta     Lion Air  
1499              Jakarta     Lion Air  
2191              Jakarta     Lion Air  
2861              Jakarta     Lion Air  
3582              Jakarta     Lion Air  
4277              Jakarta     Lion Air

number    status codeshareStatus departure.scheduledTimeLocal  \
98    JT 153  Canceled      IsOperator       2022-06-06 09:45+08:00   
769   JT 153  Canceled      IsOperator       2022-06-07 09:45+08:00   
3510  JT 153  Canceled      IsOperator       2022-06-11 09:45+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
98      2022-06-06 09:45+08:00                 3.0                    1.0   
769     2022-06-07 09:45+08:00                 3.0                    1.0   
3510    2022-06-11 09:45+08:00                 3.0                    1.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
98               B8                 WIII                  CGK   
769              A3                 WIII                  CGK   
3510            NaN                 WIII                  CGK   

     arrival.airport.name airline.name  
98                Jakarta     Lion Air  
769               Jakarta     Lion Air  
3510              Jakarta     Lion Air

number    status codeshareStatus departure.scheduledTimeLocal  \
3028  JT 155  Canceled      IsOperator       2022-06-10 14:50+08:00   
3754  JT 155  Canceled      IsOperator       2022-06-11 14:50+08:00   
4448  JT 155  Canceled      IsOperator       2022-06-12 14:50+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
3028    2022-06-10 14:50+08:00                 3.0                    1.0   
3754    2022-06-11 14:50+08:00                 3.0                    1.0   
4448    2022-06-12 14:50+08:00                 3.0                    1.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
3028             B4                 WIII                  CGK   
3754            NaN                 WIII                  CGK   
4448            NaN                 WIII                  CGK   

     arrival.airport.name airline.name  
3028              Jakarta     Lion Air  
3754              Jakarta     Lion Air  
4448              Jakarta     Lion Air

number    status codeshareStatus departure.scheduledTimeLocal  \
320   3K 245  Canceled      IsOperator       2022-06-06 15:50+08:00   
2354  3K 245  Canceled      IsOperator       2022-06-09 15:50+08:00   
5158  3K 245  Expected      IsOperator       2022-06-13 15:50+08:00   
6017  3K 245  Departed      IsOperator       2022-06-13 15:50+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
320     2022-06-06 15:50+08:00                 1.0                    NaN   
2354    2022-06-09 15:50+08:00                 1.0                    8.0   
5158    2022-06-13 15:50+08:00                 1.0                    NaN   
6017    2022-06-13 15:50+08:00                 1.0                    8.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
320             NaN                 WADD                  DPS   
2354            NaN                 WADD                  DPS   
5158            NaN                 WADD                  DPS   
6017            C22                 WADD                  DPS   

      arrival.airport.name  airline.name  
320   Denpasar-Bali Island  Jetstar Asia  
2354  Denpasar-Bali Island  Jetstar Asia  
5158  Denpasar-Bali Island  Jetstar Asia  
6017  Denpasar-Bali Island  Jetstar Asia

number    status codeshareStatus departure.scheduledTimeLocal  \
3107  JT 151  Canceled      IsOperator       2022-06-10 19:50+08:00   
4538  JT 151  Canceled      IsOperator       2022-06-12 19:50+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
3107    2022-06-10 19:50+08:00                 3.0                    1.0   
4538    2022-06-12 19:50+08:00                 3.0                    1.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
3107            NaN                 WIII                  CGK   
4538            NaN                 WIII                  CGK   

     arrival.airport.name airline.name  
3107              Jakarta     Lion Air  
4538              Jakarta     Lion Air

number             status codeshareStatus departure.scheduledTimeLocal  \
845   VN 694           Canceled      IsOperator       2022-06-07 11:30+08:00   
2189  VN 694  CanceledUncertain      IsOperator       2022-06-09 11:30+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
845     2022-06-07 11:30+08:00                 3.0                    9.0   
2189                       NaN                 3.0                    NaN   

     departure.gate arrival.airport.icao arrival.airport.iata  \
845              B6                 VVPQ                  PQC   
2189            NaN                 VVPQ                  PQC   

     arrival.airport.name airline.name  
845            Duong Dong      Vietnam  
2189           Duong Dong      Vietnam

number             status codeshareStatus departure.scheduledTimeLocal  \
360   MH 608  CanceledUncertain      IsOperator       2022-06-06 18:10+08:00   
1061  MH 608           Canceled      IsOperator       2022-06-07 18:10+08:00   
1719  MH 608  CanceledUncertain      IsOperator       2022-06-08 18:10+08:00   
2400  MH 608  CanceledUncertain      IsOperator       2022-06-09 18:10+08:00   
3076  MH 608           Departed      IsOperator       2022-06-10 18:10+08:00   
3795  MH 608  CanceledUncertain      IsOperator       2022-06-11 18:10+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
360                        NaN                 1.0                    NaN   
1061    2022-06-07 18:10+08:00                 1.0                   12.0   
1719                       NaN                 1.0                    NaN   
2400                       NaN                 1.0                    NaN   
3076                       NaN                 1.0                    NaN   
3795                       NaN                 1.0                    NaN   

     departure.gate arrival.airport.icao arrival.airport.iata  \
360             NaN                  NaN                  NaN   
1061            C16                 WMKK                  KUL   
1719            NaN                  NaN                  NaN   
2400            NaN                  NaN                  NaN   
3076            NaN                  NaN                  NaN   
3795            NaN                  NaN                  NaN   

     arrival.airport.name airline.name  
360          Kuala Lumpur     Malaysia  
1061         Kuala Lumpur     Malaysia  
1719         Kuala Lumpur     Malaysia  
2400         Kuala Lumpur     Malaysia  
3076         Kuala Lumpur     Malaysia  
3795         Kuala Lumpur     Malaysia

number    status codeshareStatus departure.scheduledTimeLocal  \
504   PX 393  Departed      IsOperator       2022-06-06 20:50+08:00   
1176  PX 393  Canceled      IsOperator       2022-06-07 20:50+08:00   
2535  PX 393  Departed      IsOperator       2022-06-09 20:35+08:00   
3250  PX 393  Departed      IsOperator       2022-06-10 20:50+08:00   
4650  PX 393  Boarding      IsOperator       2022-06-12 20:50+08:00   
5410  PX 393  Expected      IsOperator       2022-06-13 20:50+08:00   
5611  PX 393  Departed      IsOperator       2022-06-12 20:50+08:00   
6223  PX 393  Departed      IsOperator       2022-06-13 20:50+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
504     2022-06-06 23:40+08:00                 1.0                   11.0   
1176    2022-06-07 20:50+08:00                 1.0                   11.0   
2535    2022-06-09 20:35+08:00                 1.0                   11.0   
3250    2022-06-10 20:35+08:00                 1.0                   11.0   
4650    2022-06-12 20:50+08:00                 1.0                   11.0   
5410    2022-06-13 20:50+08:00                 1.0                    NaN   
5611    2022-06-12 20:50+08:00                 1.0                   11.0   
6223    2022-06-13 20:50+08:00                 1.0                   11.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
504             D36                 AYPY                  POM   
1176            NaN                 AYPY                  POM   
2535            D36                 AYPY                  POM   
3250             C1                 AYPY                  POM   
4650            D47                 AYPY                  POM   
5410            NaN                 AYPY                  POM   
5611            D47                 AYPY                  POM   
6223            C26                 AYPY                  POM   

     arrival.airport.name airline.name  
504          Port Moresby  Air Niugini  
1176         Port Moresby  Air Niugini  
2535         Port Moresby  Air Niugini  
3250         Port Moresby  Air Niugini  
4650         Port Moresby  Air Niugini  
5410         Port Moresby  Air Niugini  
5611         Port Moresby  Air Niugini  
6223         Port Moresby  Air Niugini

number    status codeshareStatus departure.scheduledTimeLocal  \
1845  SQ 522R  Canceled      IsOperator       2022-06-08 23:30+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
1845    2022-06-08 23:30+08:00                 3.0                    3.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
1845            NaN                 VOHS                  HYD   

     arrival.airport.name airline.name  
1845            Hyderabad    Singapore

number    status codeshareStatus departure.scheduledTimeLocal  \
1590  SQ 922  Departed      IsOperator       2022-06-08 07:50+08:00   
2083  SQ 922  Canceled      IsOperator       2022-06-09 07:50+08:00   
2959  SQ 922  Departed      IsOperator       2022-06-10 07:50+08:00   
3673  SQ 922  Departed      IsOperator       2022-06-11 07:50+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
1590                       NaN                 3.0                    3.0   
2083    2022-06-09 07:50+08:00                 3.0                    3.0   
2959    2022-06-10 07:50+08:00                 3.0                    3.0   
3673                       NaN                 3.0                    3.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
1590            A21                 WARR                  SUB   
2083            NaN                 WARR                  SUB   
2959            A19                 WARR                  SUB   
3673            A18                 WARR                  SUB   

     arrival.airport.name airline.name  
1590             Surabaya    Singapore  
2083             Surabaya    Singapore  
2959             Surabaya    Singapore  
3673             Surabaya    Singapore

number    status codeshareStatus departure.scheduledTimeLocal  \
1373  UB 1004  Departed      IsOperator       2022-06-08 03:45+08:00   
2733  UB 1004  Canceled      IsOperator       2022-06-10 03:45+08:00   
3443  UB 1004  Departed      IsOperator       2022-06-11 03:45+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
1373    2022-06-08 03:45+08:00                 3.0                   11.0   
2733    2022-06-10 03:45+08:00                 3.0                    NaN   
3443                       NaN                 0.0                   11.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
1373            A20                 VYYY                  RGN   
2733            NaN                 VYYY                  RGN   
3443            A12                 VYYY                  RGN   

     arrival.airport.name      airline.name  
1373               Yangon  Myanmar National  
2733               Yangon  Myanmar National  
3443               Yangon  Myanmar National

number    status codeshareStatus departure.scheduledTimeLocal  \
1198  AF 257  Departed      IsOperator       2022-06-07 22:35+08:00   
1876  AF 257  Departed      IsOperator       2022-06-08 22:40+08:00   
2883  AF 257  Canceled      IsOperator       2022-06-09 22:35+08:00   
3263  AF 257  Departed      IsOperator       2022-06-10 22:35+08:00   
4678  AF 257  Expected      IsOperator       2022-06-12 22:35+08:00   
5626  AF 257  Departed      IsOperator       2022-06-12 22:35+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
1198    2022-06-07 22:35+08:00                 1.0                    1.0   
1876    2022-06-08 22:40+08:00                 1.0                    1.0   
2883    2022-06-10 11:25+08:00                 1.0                    1.0   
3263    2022-06-10 22:35+08:00                 1.0                    1.0   
4678    2022-06-12 22:35+08:00                 1.0                    1.0   
5626    2022-06-12 22:35+08:00                 1.0                    1.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
1198            D32                 LFPG                  CDG   
1876            D47                 LFPG                  CDG   
2883            NaN                 LFPG                  CDG   
3263            D34                 LFPG                  CDG   
4678            D32                 LFPG                  CDG   
5626            D32                 LFPG                  CDG   

     arrival.airport.name airline.name  
1198                Paris   Air France  
1876                Paris   Air France  
2883                Paris   Air France  
3263                Paris   Air France  
4678                Paris   Air France  
5626                Paris   Air France

number    status codeshareStatus departure.scheduledTimeLocal  \
598   VN 658  Departed      IsOperator       2022-06-06 15:50+08:00   
1265  VN 658  Departed      IsOperator       2022-06-07 15:50+08:00   
1953  VN 658  Departed      IsOperator       2022-06-08 15:50+08:00   
2631  VN 658  Departed      IsOperator       2022-06-09 15:50+08:00   
3336  VN 658  Departed      IsOperator       2022-06-10 15:50+08:00   
4035  VN 658  Canceled      IsOperator       2022-06-11 15:50+08:00   
4747  VN 658  Departed      IsOperator       2022-06-12 15:50+08:00   
5475  VN 658  Expected      IsOperator       2022-06-13 15:50+08:00   
6321  VN 658  Departed      IsOperator       2022-06-13 15:50+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
598     2022-06-06 15:50+08:00                 3.0                    9.0   
1265    2022-06-07 15:50+08:00                 3.0                    9.0   
1953    2022-06-08 15:50+08:00                 3.0                    9.0   
2631    2022-06-09 15:50+08:00                 3.0                    9.0   
3336    2022-06-10 15:50+08:00                 3.0                    9.0   
4035    2022-06-11 15:50+08:00                 3.0                    NaN   
4747    2022-06-12 15:50+08:00                 3.0                    9.0   
5475    2022-06-13 15:50+08:00                 3.0                    NaN   
6321    2022-06-13 15:50+08:00                 3.0                    9.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
598             A17                 VVTS                  SGN   
1265            A15                 VVTS                  SGN   
1953             B9                 VVTS                  SGN   
2631            A21                 VVTS                  SGN   
3336            A11                 VVTS                  SGN   
4035            NaN                 VVTS                  SGN   
4747            A16                 VVTS                  SGN   
5475            NaN                 VVTS                  SGN   
6321             B1                 VVTS                  SGN   

     arrival.airport.name airline.name  
598      Ho Chi Minh City      Vietnam  
1265     Ho Chi Minh City      Vietnam  
1953     Ho Chi Minh City      Vietnam  
2631     Ho Chi Minh City      Vietnam  
3336     Ho Chi Minh City      Vietnam  
4035     Ho Chi Minh City      Vietnam  
4747     Ho Chi Minh City      Vietnam  
5475     Ho Chi Minh City      Vietnam  
6321     Ho Chi Minh City      Vietnam

number    status codeshareStatus departure.scheduledTimeLocal  \
556   BG 585  Departed      IsOperator       2022-06-06 15:50+08:00   
1232  BG 585  Departed      IsOperator       2022-06-07 15:50+08:00   
2590  BG 585  Departed      IsOperator       2022-06-09 15:50+08:00   
3293  BG 585  Departed      IsOperator       2022-06-10 15:50+08:00   
3990  BG 585  Departed      IsOperator       2022-06-11 15:50+08:00   
4454  BG 585  Canceled      IsOperator       2022-06-12 15:50+08:00   
5432  BG 585  Expected      IsOperator       2022-06-13 15:50+08:00   
6278  BG 585  Departed      IsOperator       2022-06-13 15:50+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
556     2022-06-06 15:50+08:00                 3.0                    5.0   
1232    2022-06-07 15:50+08:00                 3.0                    5.0   
2590    2022-06-09 15:50+08:00                 3.0                    5.0   
3293    2022-06-10 15:50+08:00                 3.0                    5.0   
3990    2022-06-11 15:50+08:00                 3.0                    5.0   
4454    2022-06-12 15:50+08:00                 3.0                    5.0   
5432    2022-06-13 15:50+08:00                 3.0                    NaN   
6278    2022-06-13 15:50+08:00                 3.0                    5.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
556              B6                 VGHS                  DAC   
1232            A11                 VGHS                  DAC   
2590            A12                 VGHS                  DAC   
3293             A9                 VGHS                  DAC   
3990             B1                 VGHS                  DAC   
4454             B7                 VGHS                  DAC   
5432            NaN                 VGHS                  DAC   
6278             B6                 VGHS                  DAC   

     arrival.airport.name      airline.name  
556                 Dhaka  Biman Bangladesh  
1232                Dhaka  Biman Bangladesh  
2590                Dhaka  Biman Bangladesh  
3293                Dhaka  Biman Bangladesh  
3990                Dhaka  Biman Bangladesh  
4454                Dhaka  Biman Bangladesh  
5432                Dhaka  Biman Bangladesh  
6278                Dhaka  Biman Bangladesh

number             status codeshareStatus departure.scheduledTimeLocal  \
1007  MH 632  CanceledUncertain         Unknown       2022-06-07 14:50+08:00   
2668  MH 632           Departed      IsOperator       2022-06-09 14:50+08:00   
4063  MH 632           Departed      IsOperator       2022-06-11 14:50+08:00   
4786  MH 632           Departed      IsOperator       2022-06-12 14:50+08:00   
6011  MH 632           Canceled      IsOperator       2022-06-13 14:50+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
1007                       NaN                 1.0                    NaN   
2668    2022-06-09 14:50+08:00                 1.0                   12.0   
4063    2022-06-11 14:50+08:00                 1.0                   12.0   
4786    2022-06-12 14:50+08:00                 1.0                   12.0   
6011    2022-06-13 14:50+08:00                 1.0                    NaN   

     departure.gate arrival.airport.icao arrival.airport.iata  \
1007            NaN                  NaN                  NaN   
2668            D36                 WMKP                  PEN   
4063             C1                 WMKP                  PEN   
4786            D32                 WMKP                  PEN   
6011            NaN                 WMKP                  PEN   

     arrival.airport.name airline.name  
1007               Penang     Malaysia  
2668               Penang     Malaysia  
4063               Penang     Malaysia  
4786               Penang     Malaysia  
6011               Penang     Malaysia

In [15]:
df[df['status']=='Canceled']

number    status codeshareStatus departure.scheduledTimeLocal  \
98     JT 153  Canceled      IsOperator       2022-06-06 09:45+08:00   
320    3K 245  Canceled      IsOperator       2022-06-06 15:50+08:00   
769    JT 153  Canceled      IsOperator       2022-06-07 09:45+08:00   
845    VN 694  Canceled      IsOperator       2022-06-07 11:30+08:00   
847    JT 163  Canceled      IsOperator       2022-06-07 12:00+08:00   
1061   MH 608  Canceled      IsOperator       2022-06-07 18:10+08:00   
1176   PX 393  Canceled      IsOperator       2022-06-07 20:50+08:00   
1499   JT 163  Canceled      IsOperator       2022-06-08 12:00+08:00   
1845  SQ 522R  Canceled      IsOperator       2022-06-08 23:30+08:00   
2083   SQ 922  Canceled      IsOperator       2022-06-09 07:50+08:00   
2354   3K 245  Canceled      IsOperator       2022-06-09 15:50+08:00   
2733  UB 1004  Canceled      IsOperator       2022-06-10 03:45+08:00   
2861   JT 163  Canceled      IsOperator       2022-06-10 12:00+08:00   
2883   AF 257  Canceled      IsOperator       2022-06-09 22:35+08:00   
3028   JT 155  Canceled      IsOperator       2022-06-10 14:50+08:00   
3107   JT 151  Canceled      IsOperator       2022-06-10 19:50+08:00   
3510   JT 153  Canceled      IsOperator       2022-06-11 09:45+08:00   
3582   JT 163  Canceled      IsOperator       2022-06-11 12:00+08:00   
3754   JT 155  Canceled      IsOperator       2022-06-11 14:50+08:00   
4035   VN 658  Canceled      IsOperator       2022-06-11 15:50+08:00   
4277   JT 163  Canceled      IsOperator       2022-06-12 12:00+08:00   
4448   JT 155  Canceled      IsOperator       2022-06-12 14:50+08:00   
4454   BG 585  Canceled      IsOperator       2022-06-12 15:50+08:00   
4538   JT 151  Canceled      IsOperator       2022-06-12 19:50+08:00   
6011   MH 632  Canceled      IsOperator       2022-06-13 14:50+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
98      2022-06-06 09:45+08:00                 3.0                    1.0   
320     2022-06-06 15:50+08:00                 1.0                    NaN   
769     2022-06-07 09:45+08:00                 3.0                    1.0   
845     2022-06-07 11:30+08:00                 3.0                    9.0   
847     2022-06-07 12:00+08:00                 3.0                    1.0   
1061    2022-06-07 18:10+08:00                 1.0                   12.0   
1176    2022-06-07 20:50+08:00                 1.0                   11.0   
1499    2022-06-08 12:00+08:00                 3.0                    1.0   
1845    2022-06-08 23:30+08:00                 3.0                    3.0   
2083    2022-06-09 07:50+08:00                 3.0                    3.0   
2354    2022-06-09 15:50+08:00                 1.0                    8.0   
2733    2022-06-10 03:45+08:00                 3.0                    NaN   
2861    2022-06-10 12:00+08:00                 3.0                    1.0   
2883    2022-06-10 11:25+08:00                 1.0                    1.0   
3028    2022-06-10 14:50+08:00                 3.0                    1.0   
3107    2022-06-10 19:50+08:00                 3.0                    1.0   
3510    2022-06-11 09:45+08:00                 3.0                    1.0   
3582    2022-06-11 12:00+08:00                 3.0                    1.0   
3754    2022-06-11 14:50+08:00                 3.0                    1.0   
4035    2022-06-11 15:50+08:00                 3.0                    NaN   
4277    2022-06-12 12:00+08:00                 3.0                    1.0   
4448    2022-06-12 14:50+08:00                 3.0                    1.0   
4454    2022-06-12 15:50+08:00                 3.0                    5.0   
4538    2022-06-12 19:50+08:00                 3.0                    1.0   
6011    2022-06-13 14:50+08:00                 1.0                    NaN   

     departure.gate arrival.airport.icao arrival.airport.iata  \
98               B8                 WIII                  CGK   
3

The conclusion for unknown flights are the same as canceled uncertain flights. Canceled flights in this case actually have all departed. Would require checks as this might not be the same every time.

Expected

Boarding               

CanceledUncertain - Checked   
Canceled - Checked

GateClosed            

Delayed

Unknown - Checked


In [16]:
df[df['status']=='Delayed']['number'].value_counts()

ZG 54      1
AY 6611    1
EY 2786    1
FY 7328    1
KL 4092    1
SQ 8386    1
SQ 5610    1
WY 5605    1
TR 734     1
3K 765     1
MH 610     1
Name: number, dtype: int64

In [17]:
tempList = []
for i in list(df[df['status']=='Delayed']['number'].value_counts().index):
    for group in df[df['number']==i].groupby('departure.scheduledTimeLocal'):
        if 'Delayed' in list(group[1]['status']):
            #display((group[1]))
            tempList.append(list(group[1]['status']))
print(pd.Series(tempList).value_counts())

[Delayed, Departed]    11
dtype: int64


In [18]:
tempList = []
for i in list(df[df['status']=='GateClosed']['number'].value_counts().index):
    for group in df[df['number']==i].groupby('departure.scheduledTimeLocal'):
        if 'GateClosed' in list(group[1]['status']):
            #display((group[1]))
            tempList.append(list(group[1]['status']))
print(pd.Series(tempList).value_counts())

[GateClosed, Departed]    17
dtype: int64


In [19]:
tempList = []
for i in list(df[df['status']=='Boarding']['number'].value_counts().index):
    for group in df[df['number']==i].groupby('departure.scheduledTimeLocal'):
        if 'Boarding' in list(group[1]['status']):
            #display((group[1]))
            tempList.append(list(group[1]['status']))
print(pd.Series(tempList).value_counts())

[Boarding, Departed]    46
[Boarding]               1
dtype: int64


In [20]:
tempList = []
for i in list(df[df['status']=='Expected']['number'].value_counts().index):
    for group in df[df['number']==i].groupby('departure.scheduledTimeLocal'):
        if 'Expected' in list(group[1]['status']):
            #display((group[1]))
            tempList.append(list(group[1]['status']))
print(pd.Series(tempList).value_counts())

[Expected, Departed]    772
[Expected]               10
dtype: int64


The codes above checks the flight numbers that has the statuses 'Delayed','GateClosed','Boarding','Expected' and group them by planned departure timing. We can see from the results that we have a unique row for "Boarding" and 10 unique rows for "Expected", the rest of the flights has subsequently departed and can be removed.

In [21]:
tempList = []
for i in list(df[df['status']=='Boarding']['number'].value_counts().index):
    for group in df[df['number']==i].groupby('departure.scheduledTimeLocal'):
        if 'Boarding' in list(group[1]['status']):
            #display((group[1]))
            if len(list(group[1]['status']))==1:
                tempList.append(i)
df[df['number'] == tempList[0]]

number    status codeshareStatus departure.scheduledTimeLocal  \
2355  7C 9990  Boarding      IsOperator       2022-06-09 15:30+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
2355    2022-06-09 15:30+08:00                 1.0                   14.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
2355            D41                 RKPK                  PUS   

     arrival.airport.name airline.name  
2355             Busan-si     Jeju Air

In [22]:
tempList = []
for i in list(df[df['status']=='Expected']['number'].value_counts().index):
    for group in df[df['number']==i].groupby('departure.scheduledTimeLocal'):
        if 'Expected' in list(group[1]['status']):
            #display((group[1]))
            if len(list(group[1]['status']))==1:
                tempList.append(i)
for i in tempList:
    display(df[df['number'] == i])

number    status codeshareStatus departure.scheduledTimeLocal  \
502   SQ 26  Departed      IsOperator       2022-06-06 23:55+08:00   
1178  SQ 26  Departed      IsOperator       2022-06-07 23:55+08:00   
1852  SQ 26  Departed      IsOperator       2022-06-08 23:55+08:00   
2533  SQ 26  Departed      IsOperator       2022-06-09 23:55+08:00   
3236  SQ 26  Departed      IsOperator       2022-06-10 23:55+08:00   
3932  SQ 26  Departed      IsOperator       2022-06-11 23:55+08:00   
4652  SQ 26  Expected      IsOperator       2022-06-12 23:55+08:00   
5391  SQ 26  Expected      IsOperator       2022-06-13 23:55+08:00   
5852  SQ 26  Departed      IsOperator       2022-06-12 23:55+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
502     2022-06-06 23:55+08:00                 3.0                    3.0   
1178    2022-06-07 23:55+08:00                 3.0                    3.0   
1852    2022-06-08 23:55+08:00                 3.0                    3.0   
2533    2022-06-09 23:55+08:00                 3.0                    3.0   
3236    2022-06-10 23:55+08:00                 3.0                    3.0   
3932    2022-06-11 23:55+08:00                 3.0                    3.0   
4652    2022-06-12 23:55+08:00                 3.0                    3.0   
5391    2022-06-13 23:55+08:00                 3.0                    NaN   
5852    2022-06-13 00:10+08:00                 3.0                    3.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
502              A2                 EDDF                  FRA   
1178             A4                 EDDF                  FRA   
1852            A10                 EDDF                  FRA   
2533             A2                 EDDF                  FRA   
3236             A2                 EDDF                  FRA   
3932             B2                 EDDF                  FRA   
4652             A5                 EDDF                  FRA   
5391            NaN                 EDDF                  FRA   
5852             A5                 EDDF                  FRA   

     arrival.airport.name airline.name  
502     Frankfurt-am-Main    Singapore  
1178    Frankfurt-am-Main    Singapore  
1852    Frankfurt-am-Main    Singapore  
2533    Frankfurt-am-Main    Singapore  
3236    Frankfurt-am-Main    Singapore  
3932    Frankfurt-am-Main    Singapore  
4652    Frankfurt-am-Main    Singapore  
5391    Frankfurt-am-Main    Singapore  
5852    Frankfurt-am-Main    Singapore

number    status codeshareStatus departure.scheduledTimeLocal  \
497   TP 8416  Departed    IsCodeshared       2022-06-06 23:55+08:00   
1171  TP 8416  Departed    IsCodeshared       2022-06-07 23:55+08:00   
1847  TP 8416  Departed    IsCodeshared       2022-06-08 23:55+08:00   
2529  TP 8416  Departed    IsCodeshared       2022-06-09 23:55+08:00   
3231  TP 8416  Departed    IsCodeshared       2022-06-10 23:55+08:00   
3929  TP 8416  Departed    IsCodeshared       2022-06-11 23:55+08:00   
4646  TP 8416  Expected    IsCodeshared       2022-06-12 23:55+08:00   
5385  TP 8416  Expected    IsCodeshared       2022-06-13 23:55+08:00   
5674  TP 8416  Departed    IsCodeshared       2022-06-12 23:55+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
497     2022-06-06 23:55+08:00                 3.0                    3.0   
1171    2022-06-07 23:55+08:00                 3.0                    3.0   
1847    2022-06-08 23:55+08:00                 3.0                    3.0   
2529    2022-06-09 23:55+08:00                 3.0                    3.0   
3231    2022-06-10 23:55+08:00                 3.0                    3.0   
3929    2022-06-11 23:55+08:00                 3.0                    3.0   
4646    2022-06-12 23:55+08:00                 3.0                    3.0   
5385    2022-06-13 23:55+08:00                 3.0                    NaN   
5674    2022-06-13 00:10+08:00                 3.0                    3.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
497              A2                  NaN                  NaN   
1171             A4                  NaN                  NaN   
1847            A10                  NaN                  NaN   
2529             A2                  NaN                  NaN   
3231             A2                  NaN                  NaN   
3929             B2                  NaN                  NaN   
4646             A5                  NaN                  NaN   
5385            NaN                  NaN                  NaN   
5674             A5                  NaN                  NaN   

     arrival.airport.name      airline.name  
497             Frankfurt  TAP Air Portugal  
1171            Frankfurt  TAP Air Portugal  
1847            Frankfurt  TAP Air Portugal  
2529            Frankfurt  TAP Air Portugal  
3231            Frankfurt  TAP Air Portugal  
3929            Frankfurt  TAP Air Portugal  
4646            Frankfurt  TAP Air Portugal  
5385            Frankfurt  TAP Air Portugal  
5674            Frankfurt  TAP Air Portugal

number    status codeshareStatus departure.scheduledTimeLocal  \
483   FJ 5903  Departed    IsCodeshared       2022-06-06 23:55+08:00   
1158  FJ 5903  Departed    IsCodeshared       2022-06-07 23:55+08:00   
1829  FJ 5903  Departed    IsCodeshared       2022-06-08 23:55+08:00   
2521  FJ 5903  Departed    IsCodeshared       2022-06-09 23:55+08:00   
3213  FJ 5903  Departed    IsCodeshared       2022-06-10 23:55+08:00   
3912  FJ 5903  Departed    IsCodeshared       2022-06-11 23:55+08:00   
4642  FJ 5903  Expected    IsCodeshared       2022-06-12 23:55+08:00   
5361  FJ 5903  Expected    IsCodeshared       2022-06-13 23:55+08:00   
5668  FJ 5903  Departed    IsCodeshared       2022-06-12 23:55+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
483     2022-06-06 23:55+08:00                 3.0                    3.0   
1158    2022-06-07 23:55+08:00                 3.0                    3.0   
1829    2022-06-08 23:55+08:00                 3.0                    3.0   
2521    2022-06-09 23:55+08:00                 3.0                    3.0   
3213    2022-06-10 23:55+08:00                 3.0                    3.0   
3912    2022-06-11 23:55+08:00                 3.0                    3.0   
4642    2022-06-12 23:55+08:00                 3.0                    3.0   
5361    2022-06-13 23:55+08:00                 3.0                    NaN   
5668    2022-06-13 00:10+08:00                 3.0                    3.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
483              A2                  NaN                  NaN   
1158             A4                  NaN                  NaN   
1829            A10                  NaN                  NaN   
2521             A2                  NaN                  NaN   
3213             A2                  NaN                  NaN   
3912             B2                  NaN                  NaN   
4642             A5                  NaN                  NaN   
5361            NaN                  NaN                  NaN   
5668             A5                  NaN                  NaN   

     arrival.airport.name  airline.name  
483             Frankfurt  Fiji Airways  
1158            Frankfurt  Fiji Airways  
1829            Frankfurt  Fiji Airways  
2521            Frankfurt  Fiji Airways  
3213            Frankfurt  Fiji Airways  
3912            Frankfurt  Fiji Airways  
4642            Frankfurt  Fiji Airways  
5361            Frankfurt  Fiji Airways  
5668            Frankfurt  Fiji Airways

number    status codeshareStatus departure.scheduledTimeLocal  \
491   OU 5809  Departed    IsCodeshared       2022-06-06 23:55+08:00   
1166  OU 5809  Departed    IsCodeshared       2022-06-07 23:55+08:00   
1839  OU 5809  Departed    IsCodeshared       2022-06-08 23:55+08:00   
2527  OU 5809  Departed    IsCodeshared       2022-06-09 23:55+08:00   
3220  OU 5809  Departed    IsCodeshared       2022-06-10 23:55+08:00   
3923  OU 5809  Departed    IsCodeshared       2022-06-11 23:55+08:00   
4641  OU 5809  Expected    IsCodeshared       2022-06-12 23:55+08:00   
5374  OU 5809  Expected    IsCodeshared       2022-06-13 23:55+08:00   
5671  OU 5809  Departed    IsCodeshared       2022-06-12 23:55+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
491     2022-06-06 23:55+08:00                 3.0                    3.0   
1166    2022-06-07 23:55+08:00                 3.0                    3.0   
1839    2022-06-08 23:55+08:00                 3.0                    3.0   
2527    2022-06-09 23:55+08:00                 3.0                    3.0   
3220    2022-06-10 23:55+08:00                 3.0                    3.0   
3923    2022-06-11 23:55+08:00                 3.0                    3.0   
4641    2022-06-12 23:55+08:00                 3.0                    3.0   
5374    2022-06-13 23:55+08:00                 3.0                    NaN   
5671    2022-06-13 00:10+08:00                 3.0                    3.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
491              A2                  NaN                  NaN   
1166             A4                  NaN                  NaN   
1839            A10                  NaN                  NaN   
2527             A2                  NaN                  NaN   
3220             A2                  NaN                  NaN   
3923             B2                  NaN                  NaN   
4641             A5                  NaN                  NaN   
5374            NaN                  NaN                  NaN   
5671             A5                  NaN                  NaN   

     arrival.airport.name airline.name  
491             Frankfurt      Croatia  
1166            Frankfurt      Croatia  
1839            Frankfurt      Croatia  
2527            Frankfurt      Croatia  
3220            Frankfurt      Croatia  
3923            Frankfurt      Croatia  
4641            Frankfurt      Croatia  
5374            Frankfurt      Croatia  
5671            Frankfurt      Croatia

number    status codeshareStatus departure.scheduledTimeLocal  \
676   NZ 3324  Departed    IsCodeshared       2022-06-06 23:55+08:00   
1163  NZ 3324  Departed    IsCodeshared       2022-06-07 23:55+08:00   
1836  NZ 3324  Departed    IsCodeshared       2022-06-08 23:55+08:00   
2524  NZ 3324  Departed    IsCodeshared       2022-06-09 23:55+08:00   
3215  NZ 3324  Departed    IsCodeshared       2022-06-10 23:55+08:00   
3918  NZ 3324  Departed    IsCodeshared       2022-06-11 23:55+08:00   
4638  NZ 3324  Expected    IsCodeshared       2022-06-12 23:55+08:00   
5371  NZ 3324  Expected    IsCodeshared       2022-06-13 23:55+08:00   
5600  NZ 3324  Departed    IsCodeshared       2022-06-12 23:55+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
676     2022-06-07 00:10+08:00                 3.0                    3.0   
1163    2022-06-07 23:55+08:00                 3.0                    3.0   
1836    2022-06-08 23:55+08:00                 3.0                    3.0   
2524    2022-06-09 23:55+08:00                 3.0                    3.0   
3215    2022-06-10 23:55+08:00                 3.0                    3.0   
3918    2022-06-11 23:55+08:00                 3.0                    3.0   
4638    2022-06-12 23:55+08:00                 3.0                    3.0   
5371    2022-06-13 23:55+08:00                 3.0                    NaN   
5600    2022-06-12 23:55+08:00                 3.0                    3.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
676             C25                 EHAM                  AMS   
1163            F58                 EHAM                  AMS   
1836             A3                 EHAM                  AMS   
2524             A5                 EHAM                  AMS   
3215            A17                 EHAM                  AMS   
3918            C13                 EHAM                  AMS   
4638             B3                 EHAM                  AMS   
5371            NaN                 EHAM                  AMS   
5600             B3                 EHAM                  AMS   

     arrival.airport.name     airline.name  
676             Amsterdam  Air New Zealand  
1163            Amsterdam  Air New Zealand  
1836            Amsterdam  Air New Zealand  
2524            Amsterdam  Air New Zealand  
3215            Amsterdam  Air New Zealand  
3918            Amsterdam  Air New Zealand  
4638            Amsterdam  Air New Zealand  
5371            Amsterdam  Air New Zealand  
5600            Amsterdam  Air New Zealand

number    status codeshareStatus departure.scheduledTimeLocal  \
870   SQ 324  Departed      IsOperator       2022-06-06 23:55+08:00   
1200  SQ 324  Departed      IsOperator       2022-06-07 23:55+08:00   
1875  SQ 324  Departed      IsOperator       2022-06-08 23:55+08:00   
2559  SQ 324  Departed      IsOperator       2022-06-09 23:55+08:00   
3257  SQ 324  Departed      IsOperator       2022-06-10 23:55+08:00   
3950  SQ 324  Departed      IsOperator       2022-06-11 23:55+08:00   
4672  SQ 324  Expected      IsOperator       2022-06-12 23:55+08:00   
5402  SQ 324  Expected      IsOperator       2022-06-13 23:55+08:00   
5623  SQ 324  Departed      IsOperator       2022-06-12 23:55+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
870     2022-06-07 00:10+08:00                 3.0                    3.0   
1200    2022-06-07 23:55+08:00                 3.0                    3.0   
1875    2022-06-08 23:55+08:00                 3.0                    3.0   
2559    2022-06-09 23:55+08:00                 3.0                    3.0   
3257    2022-06-10 23:55+08:00                 3.0                    3.0   
3950    2022-06-11 23:55+08:00                 3.0                    3.0   
4672    2022-06-12 23:55+08:00                 3.0                    3.0   
5402    2022-06-13 23:55+08:00                 3.0                    NaN   
5623    2022-06-12 23:55+08:00                 3.0                    3.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
870             C25                 EHAM                  AMS   
1200            F58                 EHAM                  AMS   
1875             A3                 EHAM                  AMS   
2559             A5                 EHAM                  AMS   
3257            A17                 EHAM                  AMS   
3950            C13                 EHAM                  AMS   
4672             B3                 EHAM                  AMS   
5402            NaN                 EHAM                  AMS   
5623             B3                 EHAM                  AMS   

     arrival.airport.name airline.name  
870             Amsterdam    Singapore  
1200            Amsterdam    Singapore  
1875            Amsterdam    Singapore  
2559            Amsterdam    Singapore  
3257            Amsterdam    Singapore  
3950            Amsterdam    Singapore  
4672            Amsterdam    Singapore  
5402            Amsterdam    Singapore  
5623            Amsterdam    Singapore

number    status codeshareStatus departure.scheduledTimeLocal  \
489   NZ 3336  Departed    IsCodeshared       2022-06-06 23:55+08:00   
1164  NZ 3336  Departed    IsCodeshared       2022-06-07 23:55+08:00   
1837  NZ 3336  Departed    IsCodeshared       2022-06-08 23:55+08:00   
2525  NZ 3336  Departed    IsCodeshared       2022-06-09 23:55+08:00   
3216  NZ 3336  Departed    IsCodeshared       2022-06-10 23:55+08:00   
3919  NZ 3336  Departed    IsCodeshared       2022-06-11 23:55+08:00   
4639  NZ 3336  Expected    IsCodeshared       2022-06-12 23:55+08:00   
5372  NZ 3336  Expected    IsCodeshared       2022-06-13 23:55+08:00   
5670  NZ 3336  Departed    IsCodeshared       2022-06-12 23:55+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
489     2022-06-06 23:55+08:00                 3.0                    3.0   
1164    2022-06-07 23:55+08:00                 3.0                    3.0   
1837    2022-06-08 23:55+08:00                 3.0                    3.0   
2525    2022-06-09 23:55+08:00                 3.0                    3.0   
3216    2022-06-10 23:55+08:00                 3.0                    3.0   
3919    2022-06-11 23:55+08:00                 3.0                    3.0   
4639    2022-06-12 23:55+08:00                 3.0                    3.0   
5372    2022-06-13 23:55+08:00                 3.0                    NaN   
5670    2022-06-13 00:10+08:00                 3.0                    3.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
489              A2                  NaN                  NaN   
1164             A4                  NaN                  NaN   
1837            A10                  NaN                  NaN   
2525             A2                  NaN                  NaN   
3216             A2                  NaN                  NaN   
3919             B2                  NaN                  NaN   
4639             A5                  NaN                  NaN   
5372            NaN                  NaN                  NaN   
5670             A5                  NaN                  NaN   

     arrival.airport.name     airline.name  
489             Frankfurt  Air New Zealand  
1164            Frankfurt  Air New Zealand  
1837            Frankfurt  Air New Zealand  
2525            Frankfurt  Air New Zealand  
3216            Frankfurt  Air New Zealand  
3919            Frankfurt  Air New Zealand  
4639            Frankfurt  Air New Zealand  
5372            Frankfurt  Air New Zealand  
5670            Frankfurt  Air New Zealand

number    status codeshareStatus departure.scheduledTimeLocal  \
480   AC 6296  Departed    IsCodeshared       2022-06-06 23:55+08:00   
1152  AC 6296  Departed    IsCodeshared       2022-06-07 23:55+08:00   
1823  AC 6296  Departed    IsCodeshared       2022-06-08 23:55+08:00   
2519  AC 6296  Departed    IsCodeshared       2022-06-09 23:55+08:00   
3207  AC 6296  Departed    IsCodeshared       2022-06-10 23:55+08:00   
3905  AC 6296  Departed    IsCodeshared       2022-06-11 23:55+08:00   
4626  AC 6296  Expected    IsCodeshared       2022-06-12 23:55+08:00   
5357  AC 6296  Expected    IsCodeshared       2022-06-13 23:55+08:00   
5667  AC 6296  Departed    IsCodeshared       2022-06-12 23:55+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
480     2022-06-06 23:55+08:00                 3.0                    3.0   
1152    2022-06-07 23:55+08:00                 3.0                    3.0   
1823    2022-06-08 23:55+08:00                 3.0                    3.0   
2519    2022-06-09 23:55+08:00                 3.0                    3.0   
3207    2022-06-10 23:55+08:00                 3.0                    3.0   
3905    2022-06-11 23:55+08:00                 3.0                    3.0   
4626    2022-06-12 23:55+08:00                 3.0                    3.0   
5357    2022-06-13 23:55+08:00                 3.0                    NaN   
5667    2022-06-13 00:10+08:00                 3.0                    3.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
480              A2                  NaN                  NaN   
1152             A4                  NaN                  NaN   
1823            A10                  NaN                  NaN   
2519             A2                  NaN                  NaN   
3207             A2                  NaN                  NaN   
3905             B2                  NaN                  NaN   
4626             A5                  NaN                  NaN   
5357            NaN                  NaN                  NaN   
5667             A5                  NaN                  NaN   

     arrival.airport.name airline.name  
480             Frankfurt   Air Canada  
1152            Frankfurt   Air Canada  
1823            Frankfurt   Air Canada  
2519            Frankfurt   Air Canada  
3207            Frankfurt   Air Canada  
3905            Frankfurt   Air Canada  
4626            Frankfurt   Air Canada  
5357            Frankfurt   Air Canada  
5667            Frankfurt   Air Canada

number    status codeshareStatus departure.scheduledTimeLocal  \
485   LH 9763  Departed    IsCodeshared       2022-06-06 23:55+08:00   
1159  LH 9763  Departed    IsCodeshared       2022-06-07 23:55+08:00   
1831  LH 9763  Departed    IsCodeshared       2022-06-08 23:55+08:00   
2522  LH 9763  Departed    IsCodeshared       2022-06-09 23:55+08:00   
3224  LH 9763  Departed    IsCodeshared       2022-06-10 23:55+08:00   
3914  LH 9763  Departed    IsCodeshared       2022-06-11 23:55+08:00   
4634  LH 9763  Expected    IsCodeshared       2022-06-12 23:55+08:00   
5366  LH 9763  Expected    IsCodeshared       2022-06-13 23:55+08:00   
5669  LH 9763  Departed    IsCodeshared       2022-06-12 23:55+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
485     2022-06-06 23:55+08:00                 3.0                    3.0   
1159    2022-06-07 23:55+08:00                 3.0                    3.0   
1831    2022-06-08 23:55+08:00                 3.0                    3.0   
2522    2022-06-09 23:55+08:00                 3.0                    3.0   
3224    2022-06-10 23:55+08:00                 3.0                    3.0   
3914    2022-06-11 23:55+08:00                 3.0                    3.0   
4634    2022-06-12 23:55+08:00                 3.0                    3.0   
5366    2022-06-13 23:55+08:00                 3.0                    NaN   
5669    2022-06-13 00:10+08:00                 3.0                    3.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
485              A2                  NaN                  NaN   
1159             A4                  NaN                  NaN   
1831            A10                  NaN                  NaN   
2522             A2                  NaN                  NaN   
3224             A2                  NaN                  NaN   
3914             B2                  NaN                  NaN   
4634             A5                  NaN                  NaN   
5366            NaN                  NaN                  NaN   
5669             A5                  NaN                  NaN   

     arrival.airport.name airline.name  
485             Frankfurt    Lufthansa  
1159            Frankfurt    Lufthansa  
1831            Frankfurt    Lufthansa  
2522            Frankfurt    Lufthansa  
3224            Frankfurt    Lufthansa  
3914            Frankfurt    Lufthansa  
4634            Frankfurt    Lufthansa  
5366            Frankfurt    Lufthansa  
5669            Frankfurt    Lufthansa

number    status codeshareStatus departure.scheduledTimeLocal  \
478   A3 1201  Departed    IsCodeshared       2022-06-06 23:55+08:00   
1150  A3 1201  Departed    IsCodeshared       2022-06-07 23:55+08:00   
1821  A3 1201  Departed    IsCodeshared       2022-06-08 23:55+08:00   
2517  A3 1201  Departed    IsCodeshared       2022-06-09 23:55+08:00   
3205  A3 1201  Departed    IsCodeshared       2022-06-10 23:55+08:00   
3903  A3 1201  Departed    IsCodeshared       2022-06-11 23:55+08:00   
4624  A3 1201  Expected    IsCodeshared       2022-06-12 23:55+08:00   
5355  A3 1201  Expected    IsCodeshared       2022-06-13 23:55+08:00   
5666  A3 1201  Departed    IsCodeshared       2022-06-12 23:55+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
478     2022-06-06 23:55+08:00                 3.0                    3.0   
1150    2022-06-07 23:55+08:00                 3.0                    3.0   
1821    2022-06-08 23:55+08:00                 3.0                    3.0   
2517    2022-06-09 23:55+08:00                 3.0                    3.0   
3205    2022-06-10 23:55+08:00                 3.0                    3.0   
3903    2022-06-11 23:55+08:00                 3.0                    3.0   
4624    2022-06-12 23:55+08:00                 3.0                    3.0   
5355    2022-06-13 23:55+08:00                 3.0                    NaN   
5666    2022-06-13 00:10+08:00                 3.0                    3.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
478              A2                  NaN                  NaN   
1150             A4                  NaN                  NaN   
1821            A10                  NaN                  NaN   
2517             A2                  NaN                  NaN   
3205             A2                  NaN                  NaN   
3903             B2                  NaN                  NaN   
4624             A5                  NaN                  NaN   
5355            NaN                  NaN                  NaN   
5666             A5                  NaN                  NaN   

     arrival.airport.name airline.name  
478             Frankfurt       Aegean  
1150            Frankfurt       Aegean  
1821            Frankfurt       Aegean  
2517            Frankfurt       Aegean  
3205            Frankfurt       Aegean  
3903            Frankfurt       Aegean  
4624            Frankfurt       Aegean  
5355            Frankfurt       Aegean  
5666            Frankfurt       Aegean

The 10 "Expected" Flights may have been caused by how the data is recorded as all 10 flights are arranged as the last flight of the day @ 11:35PM

In [23]:
df[df['airline.name']=='Jeju Air']

number    status codeshareStatus departure.scheduledTimeLocal  \
2355  7C 9990  Boarding      IsOperator       2022-06-09 15:30+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
2355    2022-06-09 15:30+08:00                 1.0                   14.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
2355            D41                 RKPK                  PUS   

     arrival.airport.name airline.name  
2355             Busan-si     Jeju Air

In [24]:
df[df['number']=='7C 9990']

number    status codeshareStatus departure.scheduledTimeLocal  \
2355  7C 9990  Boarding      IsOperator       2022-06-09 15:30+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
2355    2022-06-09 15:30+08:00                 1.0                   14.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
2355            D41                 RKPK                  PUS   

     arrival.airport.name airline.name  
2355             Busan-si     Jeju Air

In [25]:
df[df['arrival.airport.name']=='Busan-si']

number    status codeshareStatus departure.scheduledTimeLocal  \
2355  7C 9990  Boarding      IsOperator       2022-06-09 15:30+08:00   

     departure.actualTimeLocal  departure.terminal  departure.checkInDesk  \
2355    2022-06-09 15:30+08:00                 1.0                   14.0   

     departure.gate arrival.airport.icao arrival.airport.iata  \
2355            D41                 RKPK                  PUS   

     arrival.airport.name airline.name  
2355             Busan-si     Jeju Air

In [26]:
df['arrival.airport.name'].value_counts()

Kuala Lumpur      581
Melbourne City    373
Sydney City       370
Jakarta           322
Bangkok           301
                 ... 
Busan-si            1
Seoul-si            1
Ipokh               1
Barcelona           1
San Antonio         1
Name: arrival.airport.name, Length: 113, dtype: int64

In [27]:
temp = 0
for i in df['arrival.airport.name'].value_counts():
    print(df['arrival.airport.name'].value_counts().index[temp],i)
    temp+=1

Kuala Lumpur 581
Melbourne City 373
Sydney City 370
Jakarta 322
Bangkok 301
London 276
Brisbane City 210
Denpasar-Bali Island 200
Ho Chi Minh City 197
Perth 184
Phuket 184
Tokyo 169
Manila 142
Penang 135
Hanoi 113
Phnom Penh 98
Mumbai 95
Doha 93
Auckland 91
Chennai 76
Seoul 76
Frankfurt-am-Main 75
Munich 73
Surabaya 67
Port Darwin 64
Frankfurt 63
New Delhi 63
Hong Kong 60
Dubai City 60
Adelaide 57
Zurich 55
Taipei 54
San Francisco 52
Malé 51
Milan 51
Colombo 47
Bangalore 45
Na Thon (Ko Samui Island) 45
Christchurch 42
Hyderabad 37
Tiruchirappally 36
Tokyo (Narita) 36
Osaka 36
Kolkata 35
Bandar Seri Begawan 32
Helsinki 30
Kuching 28
Istanbul 28
Cairns 28
Dhaka 28
Amsterdam 27
Siem Reap 27
Kota Kinabalu 27
Ipoh 27
Angeles City 26
Los Angeles 26
Da Nang 25
Nadi 24
Copenhagen 23
Paris 21
Trivandrum 20
Newark 20
Hat Yai 18
Coimbatore 18
Chiangmai 18
Krabi 18
New York 18
Langkawi 18
Yangon 17
Manchester 16
Visakhapatnam 16
Ahmedabad 15
Kochi 15
Davao City 15
Amritsar 14
Gold 13
Johannesburg 

If possible, I would have liked to look further into this to understand why this flight has such unique information.

For now, I the conclusion is that all of 'Delayed','GateClosed','Boarding' and 'Expected' will be removed.